### PR TITLE
MODOAIPMH-326: MODOAIPMH - Upgrade to RMB 33

### PR DESCRIPTION
**Pull request by IlliaDaliek**
Jira: https://issues.folio.org/browse/MODOAIPMH-326

### PURPOSE
Update RMB version

### CODE REVIEW

**Comment by IlliaDaliek** [comment]
The tests weren't updated yet in this PR, I will push the commit with tests today later. You can start to see the file changes excepting the test classes.

File: src/main/java/org/folio/oaipmh/helpers/RepositoryConfigurationUtil.java [hunk 1]
```diff
@@ -1,49 +1,44 @@
...
 
-  }
+public class RepositoryConfigurationUtil {
 
-  private static final Logger logger = LoggerFactory.getLogger(RepositoryConfigurationUtil.class);
+  private static final Logger logger = LogManager.getLogger(RepositoryConfigurationUtil.class);
 
   private static final String QUERY = "module==OAIPMH";
-
+  private static Map<String, JsonObject> configsMap = new HashMap<>();
```
Comment (bohdan-suprun) [inline src/main/java/org/folio/oaipmh/helpers/RepositoryConfigurationUtil.java:32]: Is it used anywhere?

File: src/main/java/org/folio/oaipmh/helpers/RepositoryConfigurationUtil.java [hunk 2]
```diff
@@ -79,13 +75,15 @@ private RepositoryConfigurationUtil() {
...
       });
     } catch (Exception e) {
-      logger.error("Error happened initializing mod-configurations client for {} tenant", e, tenant);
+      logger.error("Error happened initializing mod-configurations client for {} tenant.", tenant, e);
       promise.fail(e);
+    } finally {
+      client.close();
     }
     return promise.future();
   }
```
Comment (bohdan-suprun) [inline src/main/java/org/folio/oaipmh/helpers/RepositoryConfigurationUtil.java:89]:
```suggestion
    return promise.future().onComplete(notUsed ->  client.close());
  }
```

File: src/main/java/org/folio/rest/impl/OaiPmhSetImpl.java [hunk 3]
```diff
@@ -81,7 +82,7 @@ public void putOaiPmhSetsById(String id, String lang, FolioSet entity, Map<Strin
           .otherwise(this::handleException)
           .onComplete(asyncResultHandler);
       } catch (Exception e) {
-        logger.error(MANAGE_SET_BY_ID_ERROR_MESSAGE_TEMPLATE, e,"putting", id, e.getMessage());
+        logger.error(MANAGE_SET_BY_ID_ERROR_MESSAGE_TEMPLATE,"putting", id, e.getMessage(), e);
```
Comment (bohdan-suprun) [inline src/main/java/org/folio/rest/impl/OaiPmhSetImpl.java:85]:
```suggestion
        logger.error(MANAGE_SET_BY_ID_ERROR_MESSAGE_TEMPLATE, "putting", id, e.getMessage(), e);
```

File: src/main/java/org/folio/rest/impl/OaiPmhSetImpl.java [hunk 4]
```diff
@@ -111,14 +112,14 @@ public void deleteOaiPmhSetsById(String id, String lang, Map<String, String> oka
...
-        logger.info("Delete set by id with id: {}", id);
+        logger.info("Delete set by id '{}'.", id);
         setService.deleteSetById(id, getTenantId(okapiHeaders))
           .map(OaiPmhSets.DeleteOaiPmhSetsByIdResponse.respond204())
           .map(Response.class::cast)
           .otherwise(this::handleException)
           .onComplete(asyncResultHandler);
       } catch (Exception e) {
-        logger.error(MANAGE_SET_BY_ID_ERROR_MESSAGE_TEMPLATE, e,"deleting", id, e.getMessage());
+        logger.error(MANAGE_SET_BY_ID_ERROR_MESSAGE_TEMPLATE,"deleting", id, e.getMessage(), e);
```
Comment (bohdan-suprun) [inline src/main/java/org/folio/rest/impl/OaiPmhSetImpl.java:122]:
```suggestion
        logger.error(MANAGE_SET_BY_ID_ERROR_MESSAGE_TEMPLATE, "deleting", id, e.getMessage(), e);
```

File: src/test/java/org/folio/oaipmh/common/TestUtil.java [hunk 5]
```diff
@@ -1,32 +1,61 @@
...
+  private static void initTestVariables() {
+    MavenXpp3Reader reader = new MavenXpp3Reader();
+    Model model;
+    try {
+      model = reader.read(new FileReader("pom.xml"));
+      moduleNameWithVersion = model.getArtifactId() + "-" + model.getVersion();
+    } catch (IOException | XmlPullParserException ex) {
+      String errorMessage = format("Cannot initialize variable \"moduleName\". Reason: %s", ex.getMessage());
+      logger.error(errorMessage, ex);
+      throw new IllegalStateException(errorMessage, ex);
```
Comment (bohdan-suprun) [inline src/test/java/org/folio/oaipmh/common/TestUtil.java:38]: I'd not log a message when an exception is thrown.

File: src/test/java/org/folio/oaipmh/helpers/records/RecordMetadataManagerTest.java [hunk 6]
```diff
@@ -364,7 +364,7 @@ private String getJsonObjectFromFile(String path) {
       byte[] encoded = Files.readAllBytes(Paths.get(file.getPath()));
       return new String(encoded, StandardCharsets.UTF_8);
     } catch (IOException e) {
-      logger.error("Unexpected error", e);
+      logger.error("Unexpected error.", e);
```
Comment (bohdan-suprun) [inline src/test/java/org/folio/oaipmh/helpers/records/RecordMetadataManagerTest.java:367]: Is it really unexpected assuming that we caught a specific exception (IOException)

File: src/test/java/org/folio/oaipmh/service/impl/InstancesServiceImplTest.java [hunk 7]
```diff
@@ -59,23 +59,23 @@
...
   void setUpClass(Vertx vertx, VertxTestContext testContext) throws Exception {
     PostgresClientFactory.setShouldResetPool(true);
-    logger.info("Test setup starting for " + TestUtil.getModuleId());
-    PostgresClient.getInstance(vertx)
-      .startEmbeddedPostgres();
-    TestUtil.prepareDatabase(vertx, testContext, OAI_TEST_TENANT, List.of(INSTANCES, REQUEST_METADATA_LB));
+    logger.info("Test setup starting for {}.", TestUtil.getModuleId());
+    PostgresClient.setPostgresTester(new PostgresTesterContainer());
+    PostgresClient.getInstance(vertx, OAI_TEST_TENANT).startPostgresTester();
+//    TestUtil.prepareDatabase(vertx, testContext, OAI_TEST_TENANT, List.of(INSTANCES, REQUEST_METADATA_LB));
```
Comment (bohdan-suprun) [inline src/test/java/org/folio/oaipmh/service/impl/InstancesServiceImplTest.java:65]: Please remove it.

File: src/test/resources/postgres-conf-kek.json [hunk 8]
```diff
@@ -3,5 +3,6 @@
   "port" : 7777,
   "database" : "postgres",
   "username" : "oaitest_mod_oai_pmh",
-  "password" : "oaiTest"
+  "password" : "oaiTest",
```
Comment (bohdan-suprun) [inline src/test/resources/postgres-conf-kek.json:6]: is it important to rename it to `src/test/resources/postgres-conf-kek.json`

File: pom.xml [hunk 9]
```diff
@@ -277,6 +285,38 @@
...
+      <artifactId>jaxb-osgi</artifactId>
+      <version>2.3.0.1</version>
+      <scope>compile</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>postgresql</artifactId>
+      <version>1.15.3</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (bohdan-suprun) [inline pom.xml:299]: It should be already required by `postgres-testing`

File: pom.xml [hunk 10]
```diff
@@ -277,6 +285,38 @@
...
+    <dependency>
+      <groupId>org.postgresql</groupId>
+      <artifactId>postgresql</artifactId>
+      <version>42.2.20</version>
+    </dependency>
+    <dependency>
+      <groupId>org.yaml</groupId>
+      <artifactId>snakeyaml</artifactId>
+      <version>1.5</version>
+    </dependency>
```
Comment (bohdan-suprun) [inline pom.xml:319]: Why do we need these dependencies?

**Comment by bohdan-suprun** [comment]
Please also fix code smells.

File: src/main/java/org/folio/oaipmh/helpers/AbstractGetRecordsHelper.java [hunk 11]
```diff
@@ -1,46 +1,32 @@
 package org.folio.oaipmh.helpers;
 
-import io.vertx.core.Context;
-import io.vertx.core.Future;
-import io.vertx.core.Handler;
-import io.vertx.core.Promise;
-import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.*;
```
[the remark this answers is not in the document]
Comment (IlliaDaliek) [reply src/main/java/org/folio/oaipmh/helpers/AbstractGetRecordsHelper.java:3]: Done

File: src/main/java/org/folio/oaipmh/helpers/AbstractGetRecordsHelper.java [hunk 12]
```diff
@@ -1,46 +1,32 @@
...
-import java.util.Map;
-
-import static org.folio.oaipmh.Constants.GENERIC_ERROR_MESSAGE;
-import static org.folio.oaipmh.Constants.REPOSITORY_MAX_RECORDS_PER_RESPONSE;
-import static org.folio.oaipmh.Constants.REPOSITORY_SUPPRESSED_RECORDS_PROCESSING;
-import static org.folio.oaipmh.Constants.RESUMPTION_TOKEN_FLOW_ERROR;
+import java.util.*;
+
+import static org.apache.commons.lang3.StringUtils.isNotEmpty;
+import static org.folio.oaipmh.Constants.*;
```
[the remark this answers is not in the document]
Comment (IlliaDaliek) [reply src/main/java/org/folio/oaipmh/helpers/AbstractGetRecordsHelper.java:21]: Done

File: src/main/java/org/folio/oaipmh/helpers/RepositoryConfigurationUtil.java (hunk 1 above)
Comment (IlliaDaliek) [reply src/main/java/org/folio/oaipmh/helpers/RepositoryConfigurationUtil.java:32]: Sure! This map holds the configs for the processed request.

File: src/test/java/org/folio/oaipmh/service/impl/InstancesServiceImplTest.java (hunk 7 above)
Comment (IlliaDaliek) [reply src/test/java/org/folio/oaipmh/service/impl/InstancesServiceImplTest.java:65]: Removed

File: src/test/resources/postgres-conf-kek.json (hunk 8 above)
Comment (IlliaDaliek) [reply src/test/resources/postgres-conf-kek.json:6]: This file was removed.

File: pom.xml (hunk 10 above)
Comment (IlliaDaliek) [reply pom.xml:319]: We need only postgresql dependency in order to have Postgres driver. 
Without this dependency, we cannot obtain a connection from test container DB.
I removed others dependencies and replaced TestUtil.getModuleName() with ModuleName.getModuleName() from RMB package.

File: src/test/java/org/folio/oaipmh/helpers/records/RecordMetadataManagerTest.java (hunk 6 above)
Comment (IlliaDaliek) [reply src/test/java/org/folio/oaipmh/helpers/records/RecordMetadataManagerTest.java:367]: Message changed to `logger.error("Error occurred while reading the file \"{}\".", path, e);`

**Review by bohdan-suprun — APPROVED** [review]

**Review by MikhailFokanov — APPROVED** [review]
Looks good to me.

**Pull request merged**